extensions/amp-date-display/1.0/date-display.js

```diff
@@ -65,26 +86,52 @@ let EnhancedVariablesV2Def;
  * @param {!DateDisplayDef.Props} props
  * @return {PreactDef.Renderable}
  */
-export function DateDisplay(props) {
-  const {render, children} = props;
-  const data = /** @type {!JsonObject} */ (getDataForTemplate(props));
+export function DateDisplay({
+  datetime,
+  timestampMs,
+  timestampSeconds,
+  displayIn,
+  locale,
+  offsetSeconds,
+  render,
+  children,
+  ...rest
+}) {
+  const epoch = getEpoch(datetime, timestampMs, timestampSeconds);
+  const data = useMemo(
+    () => getDataForTemplate(epoch, displayIn, locale, offsetSeconds),
+    [epoch, displayIn, locale, offsetSeconds]
+  );
+
+  const rendered = useRenderer(render || DEFAULT_RENDER, data);
+  const isHtml =
+    rendered && typeof rendered == 'object' && '__html' in rendered;
+
   useResourcesNotify();
 
-  return render(data, children);
+  return (
+    <Wrapper
+      {...rest}
+      as="time"
+      datetime={data['iso']}
+      dangerouslySetInnerHTML={isHtml ? rendered : null}
+    >
+      {isHtml ? null : rendered || children}
```

Comment:
In order to use `children`, they'll have to manually pass in a `render: () => null`, which seems wrong. We should only be using a renderer and `render` if they explicitly pass one in.

Reply:
TBH I'm really not sure whether `children` here is a good idea. Seems like rather unlikely to to be useful for the component itself?

Reply:
Let's add a Storybook example for this -- is the API such that you always have to pass in `render`? If not, how should the component be used without it? Would someone want to bring their own templating library?

Reply:
Here's how I looked at this. If we have a `<DateDisplay>` component, I'd expect it to just work and display the provided date. The `render` can be passed to customize it. With this positioning, I don't see a room for `children`. We used that to defer DOM overwrite until the template is ready, but I think there are other way to do that. But API-wise it doesn't really make sense to me on the component level.

Reply:
I removed `children` for now. Instead I implemented a pretty simple system that delays rendering until the template is ready. This minimizes FOUC w/o making it the responsibility of the component itself. IMHO that looks a lot cleaner for our purposes. LMKWYT.

Reply:
Added storybooks as well.